src/middlewared/middlewared/utils/__init__.py

```diff
@@ -442,3 +442,21 @@ def get_service(self, name):
 
     def get_services(self):
         return self._services
+
+# Sync the clock
+def sync_clock():
+    DEFAULT_NTP_SERVER="freebsd.pool.ntp.org"
+    client = ntplib.NTPClient()
+    server_alive = False
+    clock = None
+    try:
+        response = client.request(DEFAULT_NTP_SERVER)
+        if response.version:
+            server_alive = True
+            clock = response.tx_time
+            
+    except Exception:
+        # Internet not available, get datetime from host
+        clock = datetime.datetime.now()
```

Comment:
Relying on system clock should be avoided, loses the entire point of doing this unless we are sure ntpd did its things.

Also odds are on first install people do not even have network configured, so this should be retried on network changes if the birth time is null.

Reply:
Make sense. Changed that.